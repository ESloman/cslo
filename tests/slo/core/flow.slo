
print("Beginning regular for loop");
for (var x = 0; x < 10; x++) {
    if (x == 2) { continue; }
    print(x);
    if (x == 5) {
        break;
    }
}

# print("Beginning for loop over list");
# for (var y in [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]) {
#     if (y == 2) { continue; }
#     print(y);
#     if (y == 5) {
#         print("break...");
#         break;
#     }
# }

print("Beginning while loop");
var z = 0;
while(z < 10) {
    if (z == 2) { z++; continue; }
    print(z);
    if (z == 5) {
        break;
    }
    z++;
}
