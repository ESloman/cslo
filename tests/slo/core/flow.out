Beginning regular for loop
0
1
3
4
5
6
7
8
9
Beginning while loop
0
1
3
4
5
6
7
8
9
